test/integration/crud/find_and_modify.test.js

```diff
@@ -9,6 +9,157 @@ describe('Find and Modify', function () {
     return setupDatabase(this.configuration);
   });
 
+  context('promise tests', function () {
+    it(
+      'Should correctly execute findOneAndDelete operation With Promises and no options passed in',
+      {
+        metadata: {
+          requires: {
+            topology: ['single', 'replicaset', 'sharded']
+          }
+        },
+
+        test: function (done) {
+          var configuration = this.configuration;
+          var client = configuration.newClient(configuration.writeConcernMax(), {
+            maxPoolSize: 1
+          });
+
+          client.connect().then(function (client) {
+            var db = client.db(configuration.db);
+            // LINE var MongoClient = require('mongodb').MongoClient,
```

Comment:
let's not keep these commented lines; it's a hold over of some sort of autogenerated docs that's definitely no longer relevant